### PR TITLE
Ssh remote

### PR DESCRIPTION
**Pull request by qais-yousef**
This is still WIP in progress but I tested it on my busybox (buildroot) based target and it works fine for me. But probably needs more testing to iron more things out. It'd be good to get initial feedback at least and maybe get more people to try it with different envs.

I created a new utils/remote to abstract talking to the target. And had to do a few fixes to deal with limitations of my busybox env. Yes, my target isn't android based :-)

### CODE REVIEW

**Comment by qais-yousef** [comment]
This PR along with PR #19 address issue #18

**Comment by joelagnel** [comment]
I agree with the direction of the patches. The simple ones can also probably be split and can go in independently since those are nice to do. Thanks!

<rant>
I hate how github doesn't number patches so one has no idea which one is the first, etc..
</rant>

**Comment by qais-yousef** [comment]
Yes github confuses me too, especially when you get used to the nice views and facilities in gerrit..

I'll split things out and push another PR for them. Meantime if you can pull this branch and test it locally to make sure it doesn't break any of your workflow that'd be great! I tried on my pixel2 and didn't notice anything. Adb connection should work as it was after this change.

Thanks!

**Comment by joelagnel** [comment]
Sure, I'll give it a spin soon, thanks

**Comment by qais-yousef** [comment]
Hi Joel. I am happier with this set of patches. It'd be nice for someone else other than me to try them out. I've used them with both android and my buildroot based systems without a problem. I fixed a few bugs in push/pull for the ssh remote.

Please review and consider merging.

Thanks!

**Comment by qais-yousef** [comment]
Hmm I just realized I might need to update the documentation too. I'll wait for your review first :-)

File: addons/build-debian-tar [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/bin/sh
 #!/system/bin/sh
```
Comment (joelagnel) [inline addons/build-debian-tar]: the second #! is not necessary?

Comment (joelagnel) [reply addons/build-debian-tar]: the second #! is not necessary?

**Comment by joelagnel** [comment]
Other than the nits already mentioned, all patches LGTM. I am testing further now. Results also in a nice clean up so I love it, thank you!

File: addons/build-debian-tar (hunk 1 above)
Comment (qais-yousef) [reply addons/build-debian-tar]: Hmm I believe I tested with that but it failed. Let me try again. I remember there was no /bin/sh in my android but now I see it (I did change the image since then). I'll test without it and if it works I'll remove it.

**Comment by qais-yousef** [comment]
>The FTP mirror commit was already merged, perhaps your PR needs to be updated with it?

A bad rebase or I forgot to update the master on my fork. I'll get it fixed in the next update. I'll write a few lines in README.md about ssh support too.

**Comment by joelagnel** [comment]
Ok, I tested these and the nits are minor, so I will merge this. Do consider providing README and taking care of the #! nit for future patches, thanks a lot!

**Comment by qais-yousef** [comment]
Great thanks!
